django_ipam/tests/base/test_api.py

```diff
@@ -63,3 +72,81 @@ def test_unvailable_request_api(self):
                                     content_type='application/json')
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data, None)
+
+    def test_create_subnet_api(self):
+        data = {
+            'subnet': '10.0.0.0/32',
+            'description': 'Test Subnet'
+        }
+        response = self.client.post(reverse('ipam:subnet_list_create'),
+                                    data=json.dumps(data),
+                                    content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(str(Subnet.objects.first()), '10.0.0.0/32')
+
+    def test_read_subnet_api(self):
+        subnet_id = self._create_subnet(dict(subnet="fdb6:21b:a477::9f7/64")).id
+        response = self.client.get(reverse('ipam:subnet', args=(subnet_id,)))
+        self.assertEqual(response.status_code, 200)
+
+    def test_update_subnet_api(self):
+        data = {
+            'description': 'Test Subnet'
+        }
+        subnet_id = self._create_subnet(dict(subnet="fdb6:21b:a477::9f7/64")).id
+        response = self.client.patch(reverse('ipam:subnet', args=(subnet_id,)),
+                                     data=json.dumps(data),
+                                     content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Subnet.objects.get(pk=subnet_id).description, 'Test Subnet')
+
+    def test_delete_subnet_api(self):
+        subnet_id = self._create_subnet(dict(subnet="10.0.0.0/32")).id
+        response = self.client.delete(reverse('ipam:subnet', args=(subnet_id,)))
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(len(Subnet.objects.all()), 0)
+
+    def test_create_ip_address_api(self):
+        subnet_id = self._create_subnet(dict(subnet="10.0.0.0/24")).id
+        data = {
+            'ip_address': '10.0.0.2',
+            'subnet': str(subnet_id),
+            'description': 'Test Subnet'
+        }
+        response = self.client.post(reverse('ipam:list_create_ip_address', args=(subnet_id,)),
+                                    data=json.dumps(data),
+                                    content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(str(IpAddress.objects.first()), '10.0.0.2')
```

Comment:
access the `ip_address` attribute because `str()` representation may be changed and would break this test